react-main/src/config/routes.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+import { Route, Switch, Redirect, Router } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { history } from '../redux/store';
+import Game from '../app/components/Game';
+import Login from '../app/components/Login';
+
+export const ROUTES = {
+  GAME: () => '/game',
+  LOGIN: () => '/'
+};
+
+function Layout(auth) {
+  return auth ? <Route path={ROUTES.GAME()} component={Game}/> : <Redirect to={ROUTES.LOGIN()}/>;
+}
+
+function AppRoutes(props) {
+  let {auth} = props;
+  return (
+    <Router history={history}>
+      <Switch>
+        <Route exact path={ROUTES.LOGIN()} component={Login}/>
+        <Route path={ROUTES.GAME()} component={()=>Layout(auth)}/>
```

Comment:
there's a render prop to pass an inline function in Route I think. Even so, you could just pass a component. For that, Layout should be a component on its on and the you could do:
`<Route path={ROUTES.GAME()} component={()=>Layout}/>`